pkg/monotime/monotime.go

```diff
@@ -0,0 +1,49 @@
+// Copyright 2016 The etcd Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+package monotime
+
+import (
+	"time"
+)
+
+type Timer interface {
+	// Since returns duration between two points of time
+	Since(uint64) time.Duration
+	// SinceBegining returns duration between now and when timer is created
+	SinceBegining() time.Duration
+	// Now returns the current time in nanoseconds
+	Now() uint64
+}
+
+func New() Timer {
+	return &monoTimer{
+		begin: Now(),
+	}
+}
+
+type monoTimer struct {
+	begin uint64
+}
+
+func (t *monoTimer) Now() uint64 {
+	return Now()
+}
+
+func (t *monoTimer) Since(start uint64) time.Duration {
```

Comment:
not sure why this is a method on `monoTimer` since it doesn't use anything from `monoTimer`